online-shop/src/app/edit-product/edit-product.component.ts

```diff
@@ -41,11 +46,11 @@ export class EditProductComponent implements OnInit, OnDestroy {
 
   getProduct() {
     const id = parseInt(<string>this.route.snapshot.paramMap.get('id'));
-    this.productSubs =  this.productsService.getProductById(id)
-      .subscribe(product => {
-        this.product = product
-        this.initializeForm()
-      });
+    this.store.dispatch(getProduct({id: id}))
```

Comment:
For properties that have the same name as the variable which their value comes from, you can write it only once. E.g., `getProduct({ id })`